File: backend/app/schemas/customs.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class CustomsShipmentItem(BaseModel):
    id: str
    bol_number: str
    shipment_date: str
    importer_name: str
    exporter_name: str
    origin_port: str
    destination_port: str
    hs_code: str
    product_desc: str
    weight_kg: float
    teu_count: float
    declared_value_usd: Optional[float] = None

class CustomsShipmentsListResponse(BaseModel):
    total_count: int
    shipments: List[CustomsShipmentItem]

class IngestCustomsRequest(BaseModel):
    bol_records: List[Dict[str, Any]]

File: frontend/src/api/customs.ts
import { fetchApi } from './client';

export interface CustomsShipmentItem {
  id: string;
  bol_number: string;
  shipment_date: string;
  importer_name: string;
  exporter_name: string;
  origin_port: string;
  destination_port: string;
  hs_code: string;
  product_desc: string;
  weight_kg: number;
  teu_count: number;
  declared_value_usd?: number;
}

export interface CustomsShipmentsListResponse {
  total_count: number;
  shipments: CustomsShipmentItem[];
}

export const getCustomsShipmentsApi = (limit: number = 50) =>
  fetchApi<CustomsShipmentsListResponse>(`/api/v1/customs/shipments?limit=${limit}`);
